anonymizer/OpenDiffix.Service/CSVDataProvider.fs

```diff
@@ -18,22 +18,123 @@ let private openCsvStream stream =
 
   csv
 
-type DataProvider(stream: TextReader) =
+type private TypeInfo = { IsEmpty: bool; IsBoolean: bool; IsInteger: bool; IsReal: bool }
+
+let private unknownTypeInfo = { IsEmpty = true; IsBoolean = true; IsInteger = true; IsReal = true }
+let private integerTypeInfo = { IsEmpty = false; IsBoolean = false; IsInteger = true; IsReal = true }
```

Comment:
this is here to handle the `RowIndex` column. I tried some alternatives, but this remains as the least obtrusive one.